mod_reforged/hooks/entity/tactical/actor.nut

```diff
@@ -203,32 +204,22 @@
 
 	q.onDeath = @(__original) function( _killer, _skill, _tile, _fatalityType )
 	{
-		if ((::Const.Faction.Player in this.m.RF_DamageReceived) && this.m.RF_DamageReceived[::Const.Faction.Player].Total / this.m.RF_DamageReceived.Total >= 0.5)
+		// Store the real killer here before switching _killer for loot purposes. This is used to fix MSU onDeathWithInfo and onOtherActorDeath functions to have the correct killer
+		this.m.RF_RealKiller = _killer;
+
+		if (_killer != null && _killer.getFaction() != ::Const.Faction.Player && (::Const.Faction.Player in this.m.RF_DamageReceived) && this.m.RF_DamageReceived[::Const.Faction.Player].Total / this.m.RF_DamageReceived.Total >= 0.5)
```

Comment:
I don't see where damage done by the `::Const.Faction.PlayerAnimals` faction is accounted for.
As far as I can see that was already missing in the first implementation

Reply:
The damage from all factions is recorded in the `RF_DamageReceived` table in our hook on `onDamageReceived` in `skill_container`.

Reply:
Oh I think I misunderstood your point. Do you mean that we don't include that here for loot drop? Yeah that is a valid point, I'll add a fix for that.